src/rdiff_backup/rpath.py

```diff
@@ -997,8 +1000,22 @@ def get_temp_rpath(self, sibling=False):
             if self.__class__._temp_file_index > 100000000:
                 log.Log("Warning: Resetting tempfile index", 2)
                 self.__class__._temp_file_index = 0
-            tf = self.append('rdiff-backup.tmp.{index:d}'.format(
-                             index=self.__class__._temp_file_index))
+
+            if tempfile.tempdir and (shutil.disk_usage(self.path).free == 0):
```

Comment:
Add a comment what you're trying to do, i.e. use the defined tempdir if there is no place left on the current file system (it assumes that the file system of the temporary directory is a different one, it should be checked that it is the case, at least that it isn't full as well).

Reply:
Do we really want to introduce more checks under that while loop? If the user picks a directory that is out of disk space it'll get an exception later on in the code mentioning this. That looks enough to me as the OS exception is pretty clear. Adding more logic here which involves reading the partitions table etc sounds overkill to me.